clang/Type.d

```diff
@@ -93,31 +115,39 @@ struct Type
         return isObjCIdType || isObjCClassType || isObjCSelType;
     }
 
-    @property bool isWideCharType ()
+    @property bool isPointer ()
     {
-        with (CXTypeKind)
-            return kind == CXType_WChar;
+        return kind == CXTypeKind.CXType_Pointer;
     }
 
-    @property bool isConst ()
+    @property bool isTypedef ()
     {
-        return clang_isConstQualifiedType(cx) == 1;
+        return kind == CXTypeKind.CXType_Typedef;
     }
 
-    @property bool isExposed ()
+    @property bool isValid ()
     {
-        return kind != CXTypeKind.CXType_Unexposed;
+        return kind != CXTypeKind.CXType_Invalid;
     }
 
-    @property bool isAnonymous ()
+    @property bool isWideCharType ()
     {
-        return spelling.isEmpty;
+        return kind == CXTypeKind.CXType_WChar;
+    }
+
+    @property Type canonical()
+    {
+        return Type(clang_getCanonicalType(cx));
```

Comment:
This will only work if the type is create from a Clang type. Perhaps adding an assertion?